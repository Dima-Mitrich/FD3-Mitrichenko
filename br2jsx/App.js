'use strict';

import React from 'react';
import ReactDOM from 'react-dom';
import Br2jsx from './components/Br2jsx'

let string = "первый<br>второй<br/>третий<br />последний"

ReactDOM.render(<Br2jsx string={string}/>, document.getElementById('mainContainer'))