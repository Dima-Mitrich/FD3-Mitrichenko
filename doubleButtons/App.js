"use strict";

import React from 'react';
import ReactDOM from 'react-dom';

import DoubleButton from './components/DoubleButton';
import { withRainbowFrame } from './components/withRainbowFrame';

ReactDOM.render(
    <DoubleButton caption1='buttonOne' caption2='buttonTwo' cbPressed={(num) => { console.log(num) }}>HERE WE ARE</DoubleButton>,
    document.getElementById('mainContainer')
)

let colors = ['red', 'orange', 'yellow', 'green', '#00BFFF', 'blue', 'purple'];
let FramedButtons = withRainbowFrame(colors)(DoubleButton);

ReactDOM.render(
    <FramedButtons caption1='buttonOne' caption2='buttonTwo' cbPressed={(num) => { console.log(num) }}>NOW WE ARE HERE</FramedButtons>,
    document.getElementById('mainContainer1')
)