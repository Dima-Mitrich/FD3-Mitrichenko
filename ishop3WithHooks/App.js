"use strict";

import React from 'react';
import ReactDOM from 'react-dom';

import MyStore from './components/MyStore';

let storeName = 'Snickerhead';
let goods1 = [
    { name: 'air max', price: 80, photoURL: 'www.airM.com', remainder: 18, id: 1 },
    { name: 'air force', price: 90, photoURL: 'www.airF.com', remainder: 7, id: 2 },
    { name: 'monarch', price: 50, photoURL: 'www.monar.com', remainder: 22, id: 3 },
    { name: 'dunk low', price: 130, photoURL: 'www.dunk.com', remainder: 19, id: 4 },
    { name: 'blazer', price: 100, photoURL: 'www.blazer.com', remainder: 3, id: 5 },
    { name: 'M2K Tekno', price: 80, photoURL: 'www.MKtekno.com', remainder: 44, id: 6 },
];

ReactDOM.render(
    <MyStore name={storeName}
        goods={goods1} />,
    document.getElementById('mainContainer')
);