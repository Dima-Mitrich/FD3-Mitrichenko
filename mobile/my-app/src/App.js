import React, { Fragment } from 'react';
import CompanyButtons from './components/CompanyButtons';
import AbonentsContainer from './components/AbonentsContainer';
import FilterButtons from './components/FiltrBtns';

let mtsClients = [
  { name: 'Ivan', lastName: 'Ivanov', middleName: 'Ivanovich', ballance: 200, id: 1 },
  { name: 'Petr', lastName: 'Petrov', middleName: 'Petrovich', ballance: 100, id: 2 },
  { name: 'Sidor', lastName: 'Sidorov', middleName: 'Sidorovich', ballance: 50, id: 3 },
  { name: 'Dmitrij', lastName: 'Dmitrijev', middleName: 'Dmitrievich', ballance: -100, id: 4 },
  { name: 'Nikolay', lastName: 'Nikolayev', middleName: 'Nikolaevich', ballance: 70, id: 5 }
];

let velcomeClients = [
  { name: 'Kristina', lastName: 'Ivanova', middleName: 'Borisovna', ballance: -200, id: 1 },
  { name: 'Sasha', lastName: 'Kostleva', middleName: 'Olegovna', ballance: 100, id: 2 },
  { name: 'Jilia', lastName: 'Poster', middleName: 'Valerjevna', ballance: -50, id: 3 },
  { name: 'Valeria', lastName: 'Zayceva', middleName: 'Alexandrovna', ballance: -170, id: 4 },
  { name: 'Evgenija', lastName: 'Skardino', middleName: 'Sergeevna', ballance: 90, id: 5 }
]

function App() {
  return (
    <Fragment>
      <CompanyButtons />
      <FilterButtons />
      <AbonentsContainer mtsClients={mtsClients} velcomeClients={velcomeClients} />
    </Fragment>
  );
}

export default App;
