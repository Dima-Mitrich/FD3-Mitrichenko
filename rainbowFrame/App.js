"use strict";

import React from 'react';
import ReactDOM from 'react-dom';

import Text from './components/Text'

let colors = ['red','orange', 'yellow','green', '#00BFFF', 'blue', 'purple'];

ReactDOM.render(
    <Text colors={colors} />,
    document.getElementById('mainContainer')
);