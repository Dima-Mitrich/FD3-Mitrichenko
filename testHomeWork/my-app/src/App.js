import React, {Fragment} from 'react';
import AbonentsContainer from './components/AbonentsContainer';
import FilterButtons from './components/FiltrBtns';

import './App.css';

export let mtsClients = [
  { name: 'Ivan', lastName: 'Ivanov', middleName: 'Ivanovich', ballance: 200, id: 1 },
  { name: 'Petr', lastName: 'Petrov', middleName: 'Petrovich', ballance: 100, id: 2 },
  { name: 'Sidor', lastName: 'Sidorov', middleName: 'Sidorovich', ballance: 50, id: 3 },
  { name: 'Dmitrij', lastName: 'Dmitrijev', middleName: 'Dmitrievich', ballance: -100, id: 4 },
  { name: 'Nikolay', lastName: 'Nikolayev', middleName: 'Nikolaevich', ballance: 70, id: 5 }
];

function App() {
  return (
    <Fragment>
      <FilterButtons />
      <AbonentsContainer mtsClients={mtsClients} />
    </Fragment>
  );
}

export default App;
