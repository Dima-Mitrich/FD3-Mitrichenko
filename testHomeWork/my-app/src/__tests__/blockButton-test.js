import React from 'react';
import App from '../App';
import renderer from 'react-test-renderer';

test('проверка кнопки фильтрации "заблокированные"', () => {

    const component = renderer.create(
        <App />
    )

    let componentTree = component.toJSON();
    expect(componentTree).toMatchSnapshot();

    const allButton = component.root.findByProps({ id: 'blockedBtn' });
    allButton.props.onClick();

    componentTree = component.toJSON();
    expect(componentTree).toMatchSnapshot();

    allButton.props.onClick();

    componentTree = component.toJSON();
    expect(componentTree).toMatchSnapshot();
});