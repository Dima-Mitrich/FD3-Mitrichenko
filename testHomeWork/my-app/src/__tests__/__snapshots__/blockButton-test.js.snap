// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`проверка кнопки фильтрации "заблокированные" 1`] = `
Array [
  <div>
    <button
      id="allBtn"
      onClick={[Function]}
    >
      Все
    </button>
    <button
      id="activeBtn"
      onClick={[Function]}
    >
      Активные
    </button>
    <button
      id="blockedBtn"
      onClick={[Function]}
    >
      Заблокированные
    </button>
  </div>,
  <div>
    <table>
      <tbody>
        <tr>
          <th>
            Фамилия
          </th>
          <th>
            Имя
          </th>
          <th>
            Отчество
          </th>
          <th>
            Баланс
          </th>
          <th>
            Статус
          </th>
          <th>
            Редактировать
          </th>
          <th>
            Удалить
          </th>
        </tr>
        <tr>
          <td>
            Ivan
          </td>
          <td>
            Ivanov
          </td>
          <td>
            Ivanovich
          </td>
          <td>
            200
          </td>
          <td>
            active
          </td>
          <td>
            <button
              id="editBtn"
              onClick={[Function]}
            >
              Редактировать
            </button>
          </td>
          <td>
            <button
              id="deleteBtn"
              onClick={[Function]}
            >
              Удалить
            </button>
          </td>
        </tr>
        <tr>
          <td>
            Petr
          </td>
          <td>
            Petrov
          </td>
          <td>
            Petrovich
          </td>
          <td>
            100
          </td>
          <td>
            active
          </td>
          <td>
            <button
              id="editBtn"
              onClick={[Function]}
            >
              Редактировать
            </button>
          </td>
          <td>
            <button
              id="deleteBtn"
              onClick={[Function]}
            >
              Удалить
            </button>
          </td>
        </tr>
        <tr>
          <td>
            Sidor
          </td>
          <td>
            Sidorov
          </td>
          <td>
            Sidorovich
          </td>
          <td>
            50
          </td>
          <td>
            active
          </td>
          <td>
            <button
              id="editBtn"
              onClick={[Function]}
            >
              Редактировать
            </button>
          </td>
          <td>
            <button
              id="deleteBtn"
              onClick={[Function]}
            >
              Удалить
            </button>
          </td>
        </tr>
        <tr>
          <td>
            Dmitrij
          </td>
          <td>
            Dmitrijev
          </td>
          <td>
            Dmitrievich
          </td>
          <td>
            -100
          </td>
          <td>
            blocked
          </td>
          <td>
            <button
              id="editBtn"
              onClick={[Function]}
            >
              Редактировать
            </button>
          </td>
          <td>
            <button
              id="deleteBtn"
              onClick={[Function]}
            >
              Удалить
            </button>
          </td>
        </tr>
        <tr>
          <td>
            Nikolay
          </td>
          <td>
            Nikolayev
          </td>
          <td>
            Nikolaevich
          </td>
          <td>
            70
          </td>
          <td>
            active
          </td>
          <td>
            <button
              id="editBtn"
              onClick={[Function]}
            >
              Редактировать
            </button>
          </td>
          <td>
            <button
              id="deleteBtn"
              onClick={[Function]}
            >
              Удалить
            </button>
          </td>
        </tr>
      </tbody>
    </table>
    <button
      onClick={[Function]}
    >
      Добавить клиента
    </button>
  </div>,
]
`;

exports[`проверка кнопки фильтрации "заблокированные" 2`] = `
Array [
  <div>
    <button
      id="allBtn"
      onClick={[Function]}
    >
      Все
    </button>
    <button
      id="activeBtn"
      onClick={[Function]}
    >
      Активные
    </button>
    <button
      id="blockedBtn"
      onClick={[Function]}
    >
      Заблокированные
    </button>
  </div>,
  <div>
    <table>
      <tbody>
        <tr>
          <th>
            Фамилия
          </th>
          <th>
            Имя
          </th>
          <th>
            Отчество
          </th>
          <th>
            Баланс
          </th>
          <th>
            Статус
          </th>
          <th>
            Редактировать
          </th>
          <th>
            Удалить
          </th>
        </tr>
        <tr>
          <td>
            Dmitrij
          </td>
          <td>
            Dmitrijev
          </td>
          <td>
            Dmitrievich
          </td>
          <td>
            -100
          </td>
          <td>
            blocked
          </td>
          <td>
            <button
              id="editBtn"
              onClick={[Function]}
            >
              Редактировать
            </button>
          </td>
          <td>
            <button
              id="deleteBtn"
              onClick={[Function]}
            >
              Удалить
            </button>
          </td>
        </tr>
      </tbody>
    </table>
    <button
      onClick={[Function]}
    >
      Добавить клиента
    </button>
  </div>,
]
`;

exports[`проверка кнопки фильтрации "заблокированные" 3`] = `
Array [
  <div>
    <button
      id="allBtn"
      onClick={[Function]}
    >
      Все
    </button>
    <button
      id="activeBtn"
      onClick={[Function]}
    >
      Активные
    </button>
    <button
      id="blockedBtn"
      onClick={[Function]}
    >
      Заблокированные
    </button>
  </div>,
  <div>
    <table>
      <tbody>
        <tr>
          <th>
            Фамилия
          </th>
          <th>
            Имя
          </th>
          <th>
            Отчество
          </th>
          <th>
            Баланс
          </th>
          <th>
            Статус
          </th>
          <th>
            Редактировать
          </th>
          <th>
            Удалить
          </th>
        </tr>
        <tr>
          <td>
            Dmitrij
          </td>
          <td>
            Dmitrijev
          </td>
          <td>
            Dmitrievich
          </td>
          <td>
            -100
          </td>
          <td>
            blocked
          </td>
          <td>
            <button
              id="editBtn"
              onClick={[Function]}
            >
              Редактировать
            </button>
          </td>
          <td>
            <button
              id="deleteBtn"
              onClick={[Function]}
            >
              Удалить
            </button>
          </td>
        </tr>
      </tbody>
    </table>
    <button
      onClick={[Function]}
    >
      Добавить клиента
    </button>
  </div>,
]
`;
